status-im/status-react
# Wallet Connect message signing

fixes #20360

### Summary

Finished the message signing user story, according to the [acceptance criteria](https://www.notion.so/I-want-to-sign-messages-from-dApps-d8be9167b5b349e9b9419bd7f0290d4b?pvs=4). 

The PR includes the following changes:
* moved the status-go rpc calls to their own namespace `...wallet-connect.rpc`
* moved the signing functions to their own namespace `...wallet-connect.signing`, similar to transactions
* using the new rpc calls for `personal_sign` and `signTypedData` 
* refactored the message signing events

### Testing notes
[Instructions for connecting to dApps](https://www.notion.so/Testing-Wallet-Connect-interactions-0739f176856848999f0ea1671e5dfbe0)

### Steps to test


- Open Status
- Log in
- Make sure you are connected to a dApp _(check the **Testing notes** above)_
- From the dApp, trigger the following events 
  - `eth_sign`
  - `personal_sign`
  - `eth_signTypedData_v4`
- Slide to authorise the signing of the data with password/biometric
- The dApp should update with a success state



### Before and after screenshots comparison

### Risk

Described potential risks and worst case scenarios.

Tick **one**:

 
status: ready
